src/promptflow/promptflow/_sdk/data/executable/main.py.jinja2

```diff
@@ -226,21 +123,44 @@ def start():
 {% endfor %}
 
         cols = st.columns(7)
-        submit_bt = cols[0].form_submit_button(label='Submit')
-        clear_bt = cols[1].form_submit_button(label='Clear')
+        if "clicked" not in st.session_state:
+            st.session_state.clicked = False
+
+        if "error_message" in st.session_state:
+            st.error(st.session_state.error_message)
+
+{% if is_chat_flow %}
+{{ ' ' * indent_level * 2 }}submit_bt = cols[0].form_submit_button(label='Chat', type='primary', on_click=click_button, disabled=st.session_state.clicked)
+{% else %}
+{{ ' ' * indent_level * 2 }}submit_bt = cols[0].form_submit_button(label='Run', type='primary', on_click=click_button, disabled=st.session_state.clicked)
+{% endif %}
+        clear_bt = cols[1].form_submit_button(label='Clear', on_click=click_button, disabled=st.session_state.clicked)
 
         if submit_bt:
+            with st.spinner("Loading..."):
+                try:
 {% for flow_input, (default_value, value_type) in flow_inputs.items() %}
 {% if value_type == "list" %}
-{{ ' ' * indent_level * 3 }}{{flow_input}} = parse_html_content({{flow_input}})
+{{ ' ' * indent_level * 5 }}{{flow_input}} = parse_html_content({{flow_input}})
```

Comment:
why do we need the indent level here?

Reply:
because sometime tab can't guarantee the indent is correct, so added indent explicitly.